File: python/23_merge_k_sorted_lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeKLists(self, lists: List[ListNode]) -> ListNode:
        temp = []
        for i in range(0, len(lists)):
            tempNode = lists[i]
            while tempNode != None:
                temp.append(tempNode.val)
                tempNode = tempNode.next
        temp = sorted(temp)
        if len(temp) == 0:
            return None
        tempNode = ListNode(temp[0])
        head = tempNode
        for i in range(1, len(temp)):
            tempNode.next = ListNode(temp[i])
            tempNode = tempNode.next
        return head

File: golang/23_merge_k_sorted_lists.go
package main

import "sort"

// ListNode Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeKLists(lists []*ListNode) *ListNode {
	var temp []int
	for i := 0; i < len(lists); i++ {
		t := lists[i]
		for t != nil {
			temp = append(temp, t.Val)
			t = t.Next
		}
	}
	if len(temp) == 0 {
		return nil
	}
	sort.Ints(temp)
	tempNode := &ListNode{
		Val:  temp[0],
		Next: nil,
	}
	head := tempNode
	for i := 1; i < len(temp); i++ {
		tempNode.Next = &ListNode{
			Val:  temp[i],
			Next: nil,
		}
		tempNode = tempNode.Next
	}
	return head
}
